Replace python script by fastreplacestring.c

IwanKaramazow opened the pull request:
Adds support for snow flake engineered replacement of a string in a file.
Verified by comparing against the checksums of the python output on all libs of esy-ocaml-project.

Commit:
Replace python script by fastreplacestring.c

andreypopp commented on src/esy/buildEjectCommand/index.js:
```diff
@@ -15,6 +15,8 @@ const outdent = require('outdent');
 const {flattenArray} = require('../Utility');
 
 const RUNTIME = fs.readFileSync(require.resolve('./runtime.sh'), 'utf8');
+const fastReplaceString = fs.readFileSync(require.resolve('../../../node_modules/.bin/fastreplacestring'));
```
Does it mean that we are going to put a compiled executable into an ejected build? I think we need to put sources there and a make rule to compile it — ejected build might be copied to another machine with another architecture on which an already compiled executable won't work.

andreypopp replied:
We are doing that for `realpath.c`.

jordwalke replied:
I like how `FastReplaceString` is in its own package, but could we read that `.c` file, and spit it out to be recompiled just like we do for `realpath.c` so that it can be ejected on host?

jordwalke commented on package.json:
```diff
@@ -17,6 +17,7 @@
     "decompress-zip": "^0.3.0",
     "defaults": "^1.0.3",
     "detect-indent": "^5.0.0",
+    "fastreplacestring": "git+https://github.com/IwanKaramazow/FastReplaceString.git",
```
Git urls are okay, but we should try to provide specific git hashes.

tekknolagi replied:
20454af is current master.

tekknolagi replied:
@IwanKaramazow stylistic nit, but some of your indentation is funky. Looks like perhaps mixed spaces and tabs?

Commit:
Read src of fastreplacestring.c, instead of compilated executable.
Ejected build might be copied to another machine with another
architecture on which an already compiled executable won't work

Commit:
Link to IwanKaramazow/FastReplaceString tag 0.0.1

andreypopp commented on src/esy/buildEjectCommand/index.js:
```diff
@@ -15,6 +15,10 @@ const outdent = require('outdent');
 const RUNTIME = fs.readFileSync(require.resolve('./runtime.sh'), 'utf8');
+const fastReplaceStringSrc = fs.readFileSync(
+    require.resolve('../../../node_modules/fastreplacestring/fastreplacestring.c'),
+  'utf8');
```
Nit: indents

But we probably just need to run prettier over here it :)

andreypopp commented:
Looks good to me!

Commit:
better indentation

jordwalke commented on package.json:
```diff
@@ -17,6 +17,7 @@
     "decompress-zip": "^0.3.0",
     "defaults": "^1.0.3",
     "detect-indent": "^5.0.0",
+    "fastreplacestring": "git+https://github.com/IwanKaramazow/FastReplaceString.git#0.0.1",
```
I would prefer to pin to git hashes, but this works too.

jordwalke commented:
Woohoo! I will cut a new release to test it.

jordwalke commented:
There was a bug when publishing to release mode (not sure what that is - but it would be good if our releases looked as similar as possible to what we test locally). (I think I have a github issue to ensure that).

tekknolagi commented:
@jordwalke what bug?

jordwalke commented:
Looks like a seg-fault when installing reason-cli with this as a dependency.
